app/routes/_gcn.circulars/circulars.lib.ts

```diff
@@ -45,8 +45,11 @@ const subjectMatchers: SubjectMatcher[] = [
   [/ZTF[.\s_-]*(\d{2}[a-z]*)/i, ([, id]) => `ZTF${id.toLowerCase()}`],
   [/HAWC[.\s_-]*(\d{6}A)/i, ([, id]) => `HAWC-${id.toUpperCase()}`],
   [
-    /LIGO\/Virgo[.\s_-]*([S|G|GW])(\d{6}[a-z]+)/i,
-    ([, flag, id]) => `LIGO/Virgo ${flag.toUpperCase()}${id.toLowerCase()}`,
+    /LIGO\/Virgo(?:\/(KAGRA))?[-_ ]?(S|G|GW)(\d{5,6}[a-z]{0,2}).*/i,
```

Comment:
```suggestion
    /((?:LIGO|Virgo|KAGRA)(?:[\/-](?:LIGO|Virgo|KAGRA))+)[-_ \s]?(S|G|GW)(\d{5,}[a-z]+)/i,
```

This has three capturing groups:

- The list of instruments (any combination of LIGO, Virgo, and KAGRA, separated by slashes or dashes
- The prefix (S, G, or GW)
- The ID

Reply:
that doesn't capture "LIGO S12345ab" It was close though. I added a case for that. I also added a case for missing the LIGO/Virgo/KAGRA all together and expecting it to be undefined.